src/index.tsx

```diff
@@ -455,6 +456,93 @@ export class App extends React.Component<any, AppState> {
     }
   };
 
+  private getElementContainingPosition = (
+    elements: readonly ExcalidrawElement[],
+    x: number,
+    y: number,
+  ) => {
+    return getElementContainingPosition(
+      this.getElementsWithZoomScale(elements),
+      x,
+      y,
+    );
+  };
+
+  private getElementAtPosition = (
+    elements: readonly ExcalidrawElement[],
+    x: number,
+    y: number,
+  ) => {
+    return getElementAtPosition(this.getElementsWithZoomScale(elements), x, y);
+  };
+
+  private isOverScrollBars = (
+    elements: readonly ExcalidrawElement[],
+    x: number,
+    y: number,
+    canvasWidth: number,
+    canvasHeight: number,
+    scrollX: number,
+    scrollY: number,
+  ) => {
+    return isOverScrollBars(
+      this.getElementsWithZoomScale(elements),
+      x,
+      y,
+      canvasWidth,
+      canvasHeight,
+      scrollX,
+      scrollY,
+    );
+  };
+
+  private getElementsWithinSelection = (
+    elements: readonly ExcalidrawElement[],
+    selection: ExcalidrawElement,
+  ) => {
+    return getElementsWithinSelection(
+      this.getElementsWithZoomScale(elements),
+      selection,
+    );
+  };
+
+  private getCommonBounds = (elements: readonly ExcalidrawElement[]) => {
+    return getCommonBounds(this.getElementsWithZoomScale(elements));
+  };
+
+  private getElementsWithZoomScale = (
+    elements: readonly ExcalidrawElement[],
+  ): ExcalidrawElement[] => {
+    return elements.map(element => getScaledElement(element, this.state.zoom));
+  };
+
+  private setIsSelectedOnElement = (
+    element: ExcalidrawElement,
+    isSelected: boolean,
+  ) => {
+    elements = elements.map(someElement => {
+      if (someElement.id === element.id) {
+        someElement.isSelected = isSelected;
+      }
+      return someElement;
+    });
+  };
+
+  private selectElement = (element: ExcalidrawElement | null) => {
+    if (element === null) {
+      return;
+    }
+    element.isSelected = true;
```

Comment:
I had to keep these because the functions where using the local reference. We might want to inline it before the call to `selectElement`.